nextflow/EVAImport/main.nf

```diff
@@ -0,0 +1,315 @@
+
+log.info """
+  Importing EVA data with the following parameters: \
+  ${command_to_run}
+"""
+
+process run_eva {
+  cpus "${params.fork}"
+
+  input:
+  val wait
+  path eva_script
+  val options
+  val merge_all_types
+  val fork
+  val sort_vf
+  val chr_synonyms
+  val remove_prefix
+  val output_file
+  
+  output: val 'ok'
+  
+  script:
+  def sort_vf_table     = sort_vf ? " --sort_vf" : ""
+  def merge_all         = merge_all_types ? " --merge_all_types" : ""
+  def use_fork          = fork ? "--fork ${fork}" : ""
+  def chr_synonyms_file = chr_synonyms ? " --chr_synonyms ${chr_synonyms}" : ""
+  def rm_prefix         = remove_prefix ? " --remove_prefix ${remove_prefix}" : ""
+  
+  """
+  perl ${eva_script} ${options} $sort_vf_table $merge_all $use_fork $chr_synonyms_file $rm_prefix --output_file ${output_file}
+  """
+}
+
+process run_variant_synonyms {
+  input:
+  val wait
+  path var_syn_script
+  val source_name
+  val species
+  val input_file
+  val registry
+  val host
+  val port
+  val dbname
+  
+  output: val 'ok'
+  
+  script:
+  
+  if(species == "sus_scrofa")
+      """
+        perl ${var_syn_script} --source_name ${source_name} --species ${species} --data_file ${input_file} --registry ${registry}
+        perl ${var_syn_script} --source_name "pig_chip" --species ${species} --registry ${registry}
+      """
+  
+  else if(species == "rattus_norvegicus")
+      """
+        perl ${var_syn_script} --source_name ${source_name} --species ${species} --data_file ${input_file} --registry ${registry}
+        perl ${var_syn_script} --source_name "rat" --species ${species} --registry ${registry} --host ${host} --port ${port} --user 'ensro' --db_name $dbname
+      """
+  else 
+      """
+        perl ${var_syn_script} --source_name ${source_name} --species ${species} --data_file ${input_file} --registry ${registry}
+      """
+}
+
+process run_variation_set {
+  input:
+  val wait
+  val my_species_set
+  val var_set_script
+  val files_path
+  val filenames
+  val species
+  val registry
+
+  output: val 'ok'
+
+  script:
+
+  def input_file = filenames.get(my_species_set)
+  """
+  perl ${var_set_script} -load_file ${files_path}${input_file} -registry ${registry} -species ${species} -variation_set ${my_species_set}
+  """
+}
+
+// Post-process variation feature sets
+// Populates variation_feature.variation_set_id
+process run_variation_set_2 {
+  input:
+  val wait
+  val var_set_script_2
+  val species
+  val registry
+
+  output: val 'ok'
+
+  script:
+
+  """
+  perl ${var_set_script_2} -registry_file ${registry} -species ${species}
+  """
+}
+
+process prepare_tables {
+  input:
+  val copy_script
+  val host
+  val dbname
+  val old_dbname
+
+  output: val 'ok'
+
+  script:
+  def new_host = "${host}-ensadmin"
```

Comment:
After reviewing https://github.com/Ensembl/ensembl-internal-variation/pull/118, I noticed the `-ensadmin` here. I do not think this is safe because when running this pipeline, we never explicitly give the user/password credentials to allow it to write the database.

I think that it would be better to get user/password parameters as other pipelines do.

When starting the EVA Nextflow pipeline, you can use the shortcut you used here to automatically define the parameters with `${host}-ensadmin details`.

Reply:
Updated the script.